### PR TITLE
[DPP-590] Allow to have an extra options parameter that will be displayed in the dropdown

### PR DESCRIPTION
**Pull request by omar-aguilar**
## What did you implement:
Implementing issue: [DPP-590](https://fng-jira.fox.com/browse/DPP-590)

## How did you implement it:
Added another prop to `react-select`, named `extraOptions` which allows to render the dropdown with these extra options plus the options coming from `loadOptions` or `options` props

This change was necessary, due the API that doesn't refresh fast enough, because as it is now newly items created within a reference list are not displayed in the dropdown results until the cache gets updated

## How can we verify it:
With the steps in https://github.com/foxbroadcasting/fng-cms/pull/364

### CODE REVIEW

File: src/Select.js [hunk 1]
```diff
@@ -901,6 +901,25 @@ class Select extends React.Component {
 	filterOptions (excludeOptions) {
 		var filterValue = this.state.inputValue;
 		var options = this.props.options || [];
+
+		var extraOptions = this.props.extraOptions || [];
+		var labelKey = this.props.labelKey;
+		if(extraOptions.length) {
```
Comment (isa-bel) [inline src/Select.js]: Can you add a space here and in line 916?

**Review by isa-bel — APPROVED** [review]

File: src/Select.js [hunk 2]
```diff
@@ -901,6 +901,25 @@ class Select extends React.Component {
...
+				if (typeof value === 'object') {
+					optionsSet.add(`${value[labelKey]}`.toLowerCase());
+				}
+			});
+			const filteredExtraOptions = extraOptions.filter((value) => {
+				if (typeof value !== 'object') return false;
+				const label = `${value[labelKey]}`.toLowerCase();
+				return !optionsSet.has(label) && label.indexOf(filterValueLower) >= 0;
+			});
+			options = options.concat(filteredExtraOptions);
```
Comment (jangelon) [inline src/Select.js]: Does this end up adding the new options to the very end? I think in this case the new option they want is probably the one they want, can we have it at the top instead?
```
options = [...filteredExtraOptions, ...options];
```

I haven't actually tried it though, so they may get resorted from this array anyways.

Comment (omar-aguilar) [reply src/Select.js]: Changed the order, as the component doesn't sort the options

**Review by aarregui — APPROVED** [review]
LGTM

**Pull request merged**